### PR TITLE
i#6660 dr$sim aliases: Add "-t drmemtrace"

### LINKED ISSUE #6660: Add aliases to drmemtrace launching: "-t drmemtrace" and "-tool"
Today all drmemtrace analysis tools are launched with a front-end named "drcachesim" and with a tool selector named "-simulator_type".  Those names can be misleading.  We should add aliases and update the docs so the invocation changes from:

```
bin64/drrun -t drcachesim -simulator_type reuse_distance ...
```

to:

```
bin64/drrun -t drmemtrace -tool reuse_distance ...
```

### PR DESCRIPTION
**Pull request by derekbruening**
Adds a tool file so "-t drmemtrace" can become the preferred way to launch the drmemtrace tool framework, as "drcachesim" is just one tool within the framework.

Updates the documentation and tests.

Renames the "drcachesim" executable to "drmemtrace_launcher".

Fixes #6660

### CODE REVIEW

**Comment by derekbruening** [comment]
a64 failure is "thread is missing exit" #6733

**Comment by abhinav92003** [comment]
> a64 failure is "thread is missing exit" #6733

This is still happening? I thought (in #6793) that #6444 was the only way it happened.

File: clients/drcachesim/CMakeLists.txt [hunk 1]
```diff
@@ -720,20 +720,31 @@ ${debugopt}\n")
 endfunction ()
 
 if (X64)
-  set(CONFIG_INSTALL ${PROJECT_BINARY_DIR}/drcachesim.drrun64)
-  set(CONFIG_BUILD ${PROJECT_BINARY_DIR}/tools/drcachesim.drrun64)
+  set(CONFIG_INSTALL ${PROJECT_BINARY_DIR}/drmemtrace.drrun64)
+  set(CONFIG_BUILD ${PROJECT_BINARY_DIR}/tools/drmemtrace.drrun64)
+  # Support the older launcher name.
```
Comment (abhinav92003) [inline clients/drcachesim/CMakeLists.txt:725]: Would be nice to xref i#6660 here and elsewhere.

**Review by abhinav92003 — APPROVED** [review]

File: clients/drcachesim/CMakeLists.txt (hunk 1 above)
Comment (derekbruening) [reply clients/drcachesim/CMakeLists.txt:725]: Done.

**Comment by derekbruening** [comment]
x32 failure is drcachesim.opcode_mix encoding size assert #6303

**Pull request merged**